tap_github/authenticator.py

```diff
@@ -161,13 +161,34 @@ def prepare_tokens(self) -> Dict[str, TokenRateLimit]:
         # Get rate_limit_buffer
         rate_limit_buffer = self._config.get("rate_limit_buffer", None)
 
-        # Dedup tokens and create a dict of TokenRateLimit
+        # Dedup tokens and test them
+        filtered_tokens = []
+        for token in list(set(available_tokens)):
+            try:
+                now = datetime.now()
+                modified_since = now.strftime("%a, %d %b %Y %H:%M:%S GMT")
+                response = requests.get(
+                    url="https://api.github.com/repos/MeltanoLabs/tap-github/readme",
+                    headers={
+                        "Authorization": f"token {token}",
+                        "If-Modified-Since": modified_since,
+                    },
+                )
+                response.raise_for_status()
+                filtered_tokens.append(token)
+            except Exception as e:
```

Comment:
Could you catch the specific exception that `requests` raises in this case instead of a generic one?